### PR TITLE
Put CVMix convective viscosity into shear term unless KPP is used

### PR DESCRIPTION
**Pull request by angus-g**
In both the legacy and non-legacy diabatic drivers, the viscosity output from CVMix convection went to the `Kv_slow` term. However, this term is only allocated if KPP is being used, leading to a segfault if `USE_KPP = False`.

The `Kv_shear` term is allocated for CVMix convection and/or KPP. In fact, the non-llegacy diabatic driver was checking to see whether KPP is being used, and putting the viscosity into the `Kv_shear` term if that is the case.

It seems as though `Kv_shear` should be the default target for the convective viscosity, and `Kv_slow` should only be used in the case that KPP is also being used. I'm not entirely sure about the difference between the intended use for these two terms however, so I don't know if this is valid!

### CODE REVIEW

**Review by Hallberg-NOAA — CHANGES_REQUESTED** [review]
At one point, visc%Kv_shear and visc%Kv_slow were the same array.  I don't exactly remember why these were split up, but it may have been to create a diagnostic of the non-shear viscosity, or it may have been that when we added the vertex-point shear mixing, visc%Kv_shear_Bu, we needed to differentiate between the two, or it may have been something else to to do with refactoring the code that sets the diffusivity.

  However, there is clearly a problem that this PR is addressing, in that when `USE_KPP=False`, but `USE_CVMix_CONVECTION=True`, an unallocated pointer is being passed to `calculate_CVMix_conv()` in an array argument at lines 756 and 1313 (which this PR corrects), and also at line 1897, which it does not.  The proposed code change will change answers, but only in cases where there would have been a segmentation fault, which must imply that no one had ever tried this particular configuration before.

  I think that the right answer here is to use `visc%Kv_shear` in all of the calls to `calculate_CVMix_conv()`. This PR corrects two such instances, at lines 756 and 1313, but undoes the correct form at line 1311. I would also like to request that the same change be made to line 1897.

  After this PR has been accepted, I think that it would be worthwhile for us to revisit how Kv_slow is being handled in other cases where `USE_KPP = .false.`, and perhaps start adding those viscosities subject to the control of a Prandtl number.  I would note, however, that initializing `visc%Kv_slow(:,:,:) = CS%Kv` at about line 312 of MOM_set_diffusivity.F90 may lead to double-counting of this background diffusivity, which is also being used as a floor value in `find_coupling_coef`, but the would have `visc%Kv_shear` added to it.

File: src/parameterizations/vertical/MOM_diabatic_driver.F90 [hunk 1]
```diff
@@ -1308,9 +1308,9 @@ subroutine diabatic_ALE(u, v, h, tv, Hml, fluxes, visc, ADp, CDp, dt, Time_end,
   if (CS%use_CVMix_conv) then
     ! Increment vertical diffusion and viscosity due to convection
     if (CS%useKPP) then
-      call calculate_CVMix_conv(h, tv, G, GV, US, CS%CVMix_conv, Hml, Kd_heat, visc%Kv_shear, Kd_aux=Kd_salt)
```
[the remark this answers is not in the document]
Comment (gustavo-marques) [reply src/parameterizations/vertical/MOM_diabatic_driver.F90:1311]: I agree with @Hallberg-NOAA. Line 1311 should not be changed. `Kv_slow` was introduced ~ 4 years ago to keep track of vertical viscosities due to "slow" processes (e.g., tidal, background, convection, etc). However, this purpose seems to have been broken when the interface to calculate_CVMix_conv was revisited ([this](https://github.com/NOAA-GFDL/MOM6/commit/61cbcc2a3a179cf54463fe886d8a4ecb324895d6) PR).
The other two modifications (lines 756 and 1313) are correct.

**Comment by angus-g** [comment]
Thanks for the reviews @Hallberg-NOAA and @gustavo-marques. It's good to know the context behind this, I was definitely just searching a bit blindly for a change that wouldn't segfault. I'll submit a revision shortly.

**Comment by marshallward** [comment]
This now looks correct to me.  Others OK with it?

File: src/parameterizations/vertical/MOM_diabatic_driver.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hallberg-NOAA) [reply src/parameterizations/vertical/MOM_diabatic_driver.F90:1311]: I agree with this revised change.

**Review by Hallberg-NOAA — APPROVED** [review]

**Comment by Hallberg-NOAA** [comment]
I am approving this PR, in that the changes that in this PR are correct. However, the conversation around this commit requested an additional similar change at about line 1897 of MOM_diabatic_driver.F90, which was not included in this PR.

More broadly, there should be an examination of the role of visc%Kv_slow and visc%Kv_shear, and whether these should continue to exist as separate arrays.

**Comment by Hallberg-NOAA** [comment]
This PR has passed TC testing and pipeline testing at https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/15842 .

**Pull request merged**